src/binary/uint.rs

```diff
@@ -73,6 +69,45 @@ impl UInt {
     }
 }
 
+/// The result of the [`magnitude_to_bytes`] function. The fields are private;
+/// users must go through `as_ref()` to see the slice that ignores leading
+/// zeros.
+///
+/// This type exists to avoid heap allocation.
+pub struct U64Magnitude {
```

Comment:
Should this `#[derive(Copy, Clone, Debug)]` at a minimum since it is effectively a smallish data type.

Reply:
Added